src/plugins/geofence/geofence_impl.cpp

```diff
@@ -183,64 +55,70 @@ void GeofenceImpl::assemble_mavlink_messages(
-                _parent->get_own_component_id(),
-                message.get(),
-                _parent->get_system_id(),
-                _parent->get_autopilot_id(),
-                _mavlink_geofence_item_messages.size(),
-                MAV_FRAME_GLOBAL_INT,
-                command,
-                0, // current
-                0, // autocontinue
-                float(polygon->points.size()), // vertex count
-                0.0f,
-                0.0f,
-                0.0f,
-                int32_t(std::round(point.latitude_deg * 1e7)),
-                int32_t(std::round(point.longitude_deg * 1e7)),
-                0.0f,
-                MAV_MISSION_TYPE_FENCE);
-            _mavlink_geofence_item_messages.push_back(message);
+            // FIXME: check if these two  make sense.
+            const uint8_t current = (sequence == 0 ? 1 : 0);
+            const uint8_t autocontinue = 0;
+            const float param1 = float(polygon->points.size());
+
+            items.push_back(
+                MAVLinkMissionTransfer::ItemInt{sequence,
+                                                MAV_FRAME_GLOBAL_INT,
+                                                command,
+                                                current,
+                                                autocontinue,
+                                                param1,
+                                                0.0f,
+                                                0.0f,
+                                                0.0f,
+                                                int32_t(std::round(point.latitude_deg * 1e7)),
+                                                int32_t(std::round(point.longitude_deg * 1e7)),
+                                                0.0f,
+                                                MAV_MISSION_TYPE_FENCE});
+            ++sequence;
         }
     }
+    return items;
 }
 
-void GeofenceImpl::timeout_happened()
-{
-    LogDebug() << "timeout happened";
-    _active = false;
-
-    report_geofence_result(_result_callback, Geofence::Result::TIMEOUT);
-}
-
-void GeofenceImpl::send_geofence_item(uint16_t seq)
-{
-    LogDebug() << "Send geofence item " << int(seq);
-    if (seq >= _mavlink_geofence_item_messages.size()) {
-        LogDebug() << "Geofence item requested out of bounds.";
-        return;
-    }
-
-    _parent->send_message(*_mavlink_geofence_item_messages.at(seq));
-}
-
-void GeofenceImpl::report_geofence_result(
-    const Geofence::result_callback_t& callback, Geofence::Result result)
+Geofence::Result GeofenceImpl::convert_result(MAVLinkMissionTransfer::Result result)
 {
-    if (callback == nullptr) {
-        LogDebug() << "Callback is not set";
-        return;
+    switch (result) {
+        case MAVLinkMissionTransfer::Result::Success:
+            return Geofence::Result::SUCCESS;
+        case MAVLinkMissionTransfer::Result::ConnectionError:
+            return Geofence::Result::ERROR; // FIXME
```

Comment:
How should those be fixed? By adding Geofence results?

Reply:
I planned to consolidate the different results in a future PR. I think the result type will be duplicated / or similar between the plugins but we will add some more error types that make sense. My goal was basically to not have any change in API for this PR but only swap out internals.